# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/CopyableCanvasRenderer.h
# Commit: 16c5dd2cfd60
# Full Hash: 16c5dd2cfd600a62c80ca17cfb35c0868e016056
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r ca8139d58d58 -r 16c5dd2cfd60 gfx/layers/CopyableCanvasRenderer.h
--- a/gfx/layers/CopyableCanvasRenderer.h	Fri Jun 12 21:53:14 2020 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,77 +0,0 @@
-/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* vim: set ts=8 sts=2 et sw=2 tw=80: */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#ifndef GFX_COPYABLECANVASRENDERER_H
-#define GFX_COPYABLECANVASRENDERER_H
-
-#include <stdint.h>  // for uint32_t
-#include "CanvasRenderer.h"
-#include "GLContextTypes.h"  // for GLContext
-#include "gfxContext.h"      // for gfxContext, etc
-#include "gfxTypes.h"
-#include "gfxPlatform.h"          // for gfxImageFormat
-#include "mozilla/Assertions.h"   // for MOZ_ASSERT, etc
-#include "mozilla/Preferences.h"  // for Preferences
-#include "mozilla/RefPtr.h"       // for RefPtr
-#include "mozilla/gfx/2D.h"       // for DrawTarget
-#include "mozilla/mozalloc.h"     // for operator delete, etc
-#include "nsISupportsImpl.h"      // for MOZ_COUNT_CTOR, etc
-
-namespace mozilla {
-
-namespace gl {
-class SharedSurface;
-}  // namespace gl
-
-namespace layers {
-
-/**
- * A shared CanvasRenderer implementation that supports copying
- * its contents into a gfxASurface using RebackSurface.
- */
-class CopyableCanvasRenderer : public CanvasRenderer {
- public:
-  CopyableCanvasRenderer();
-  virtual ~CopyableCanvasRenderer();
-
- public:
-  void Initialize(const CanvasInitializeData& aData) override;
-  bool IsDataValid(const CanvasInitializeData& aData) override;
-
-  void ClearCachedResources() override;
-  void Destroy() override;
-
-  CopyableCanvasRenderer* AsCopyableCanvasRenderer() override { return this; }
-
-  bool NeedsYFlip() const { return mOriginPos == gl::OriginPos::BottomLeft; }
-  bool HasGLContext() const { return !!mGLContext; }
-  bool IsOpaque() const { return mOpaque; }
-
-  PersistentBufferProvider* GetBufferProvider() { return mBufferProvider; }
-
-  already_AddRefed<gfx::SourceSurface> ReadbackSurface();
-
- protected:
-  RefPtr<gl::GLContext> mGLContext;
-  RefPtr<PersistentBufferProvider> mBufferProvider;
-  RefPtr<AsyncCanvasRenderer> mAsyncRenderer;
-  RefPtr<OOPCanvasRenderer> mOOPRenderer;
-
-  bool mIsAlphaPremultiplied;
-  gl::OriginPos mOriginPos;
-
-  bool mOpaque;
-
-  RefPtr<gfx::DataSourceSurface> mCachedTempSurface;
-
-  gfx::DataSourceSurface* GetTempSurface(const gfx::IntSize& aSize,
-                                         const gfx::SurfaceFormat aFormat);
-};
-
-}  // namespace layers
-}  // namespace mozilla
-
-#endif